java/vector/src/main/java/org/apache/arrow/vector/util/DecimalUtility.java

```diff
@@ -120,34 +121,25 @@ public static void writeLongToArrowBuf(long value, ArrowBuf bytebuf, int index)
    * width.
    */
   public static void writeByteArrayToArrowBuf(byte[] bytes, ArrowBuf bytebuf, int index) {
-    writeByteArrayToArrowBuf(bytes, bytebuf, index, 0);
+    writeByteArrayToArrowBufHelper(bytes, bytebuf, index);
   }
 
-  private static void writeByteArrayToArrowBuf(byte[] bytes, ArrowBuf bytebuf, int index, int padValue) {
+  private static void writeByteArrayToArrowBufHelper(byte[] bytes, ArrowBuf bytebuf, int index) {
     final int startIndex = index * DECIMAL_BYTE_LENGTH;
     if (bytes.length > DECIMAL_BYTE_LENGTH) {
       throw new UnsupportedOperationException("Decimal size greater than 16 bytes");
     }
 
     // Decimal stored as little endian, need to swap data bytes before writing to ArrowBuf
     byte[] bytesLE = new byte[bytes.length];
```

Comment:
Do you think we can make this array reusable, so there is no need to create it every time?

Reply:
If its global, it would need synchronization no? Overhead of that would be greater than the allocation/gc of this array which would be in young gen which would be fast.